barman/infofile.py

```diff
@@ -945,6 +945,24 @@ def walk_to_root(self, return_self=True):
             yield backup_info
             backup_info = backup_info.get_parent_backup_info()
 
+    def is_checksum_consistent(self):
+        """
+        Check if all backups in the chain are consistent with their checksums
+        configurations.
+
+        The backup chain is considered inconsistent if the current backup was taken with
+        ``data_checksums`` enabled and any of its ascendants were taken with it disabled.
+        It is considered consistent otherwise.
+
```

Comment:
```suggestion

    .. note::
        While this method was created to check inconsistencies in chains of one
        or more (Postgres 17+ core) incremental backups, it can be safely used
        with any Postgres version and with any Barman backup method. That is
        true because it always returns ``True`` when called for a Postgres full
        backup or for a rsync backup.

```

I think we can add this note here to make this behavior explicit.
I guess it would also avoid questions in the future, if one finds weird the following statement at the end of the backup, which doesn't check the backup method nor the Postgres version:

```python
if not backup_info.is_checksum_consistent():
```

Reply:
Yeah, e.g. someone might inspecting the code specifically for rsync backups and get confused with this line at the end, so it's good to have an explanation like that so they know it is not meant for that case.